http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java

```diff
@@ -77,6 +92,10 @@
     }
 
     private Future<Channel> acquire0(Promise<Channel> promise) {
+        if (closed) {
```

Comment:
Future improvement, we can make this a decorator to reuse this functionality for HTTP/1.1

Reply:
Sounds good